Subscriber/DoctrineToGatewayEventSubscriber.php

```diff
@@ -0,0 +1,341 @@
+        $this->logger->info(
+            'Deleting object from database',
+            [
+                'object'    => $object->getId(),
+                'entity'    => $object->getEntity()->getId(),
+                'objectData'=> $object->toArray(),
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent($eventType, ['object' => $object->toArray(), 'entity' => $this->getEntity()->getId()->toString()]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.pre.delete');
+    }
+
+    /**
+     * Creating object in database.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function prePersist(LifecycleEventArgs $args): void
+    {
+        $object = $args->getObject();
+
+        // if this subscriber only applies to certain entity types,
+        if (!$object instanceof ObjectEntity) {
+            return;
+        }
+
+        // Write the log
+        $this->logger->info(
+            'Creating object in database',
+            [
+                'object'    => $object->getId(),
+                'entity'    => $object->getEntity()->getId(),
+                'objectData'=> $object->toArray(),
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent($eventType, ['object' => $object->toArray(), 'entity' => $this->getEntity()->getId()->toString()]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.pre.create');
+    }
+
+    /**
+     * Updating object to database.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function preUpdate(LifecycleEventArgs $args): void
+    {
+        $object = $args->getObject();
+
+        // if this subscriber only applies to certain entity types,
+        if (!$object instanceof ObjectEntity) {
+            return;
+        }
+
+        // Write the log
+        $this->logger->info(
+            'Updating object to database',
+            [
+                'object'    => $object->getId(),
+                'entity'    => $object->getEntity()->getId(),
+                'objectData'=> $object->toArray(),
+            ]
+        );
+
+        // Throw the event
+        $event = new ActionEvent($eventType, ['object' => $object->toArray(), 'entity' => $this->getEntity()->getId()->toString()]);
+        $this->eventDispatcher->dispatch($event, 'commongateway.object.pre.update');
+    }
+
+    /**
+     * Deleted object from database.
+     *
+     * @param LifecycleEventArgs $args
+     *
+     * @return void
+     */
+    public function postRemove(LifecycleEventArgs $args): void
+    {
+        $object = $args->getObject();
+
+        // if this subscriber only applies to certain entity types,
+        if (!$object instanceof ObjectEntity) {
+            return;
+        }
+
+        // Write the log
+        $this->logger->info(
+            'Deleted object from database',
+            [
+            ]
```

Comment:
Maybe add some data here?

Reply:
There is no data, the object ahs been deleted. Im trying to figure out if we can still get its former id or something

Reply:
hmm, if (!$object instanceof ObjectEntity) works why not getId() :(
but check